src/main/java/com/minecolonies/entity/pathfinding/PathNavigate.java

```diff
@@ -111,13 +111,29 @@ public PathResult moveToXYZ(double x, double y, double z, double speed)
         int newY = (int) y;
         int newZ = MathHelper.floor_double(z);
 
-        if (!noPath() &&
-              destination != null &&
-              BlockPosUtil.isEqual(destination, newX, newY, newZ))
+
+        if ((destination != null
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (5) used in the expression (maximum allowed 4). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)